### PR TITLE
Everything in src

### PR DESCRIPTION
**Pull request by jcbhmr**

### CODE REVIEW

**Comment by jcbhmr** [comment]
i guess nows a good time to explain why i think this is a good idea vs the previous build.mjs approach:
1. it lets you customize individual things without editing the code-generation code (you're just editing the raw file, not the thing that _generates_ said file)
2. it means you get a clear visual of what things are there and what things arent
3. its not dependant on nodejs v20 vs nodejs v16; you still have the same count
4. it makes typescript happier when the file you `import` actually exists
5. you can write custom types like with `<T>` generics for things like `Map` whereas when autogen'd it wouldn't have `<T>` (though this could be changed with more complex build.mjs)

basically it just makes things in general easier. i still think that GENERATING THE STARTER FILES is a good idea.

also pro tip: use VS Code's search-and-replace feature which lets you grep the entire src/ folder and then use regex to replace patterns with other things. this is great for mass-editing since there's like 700 files and it'd be a real pain to change them. just use regex most of the time 😉
